explorer/frontend/src/Lib/Waypoints/Waypoints.purs

```diff
@@ -42,4 +43,8 @@ waypoint elemId handler = runEffFn3 waypointImpl elemId handler defaultWaypointO
 waypoint' :: forall eff. ElementId -> (WaypointHandler eff) -> WaypointOffset -> Eff (waypoint :: WAYPOINT | eff) Waypoint
 waypoint' = runEffFn3 waypointImpl
 
-foreign import destroy :: forall eff. Fn1 Waypoint (Eff (waypoint :: WAYPOINT | eff) Unit)
```

Comment:
@sectore I am not clear why the error was occuring at all? Isn't the above correct as well

Reply:
ps, have you tried with:
```haskell
foreign import destroy :: forall eff. Fn1 Waypoint (Eff (waypoint :: WAYPOINT | eff) Waypoint)
```

?

Reply:
The foreign function `destroy` was called directly from `Update.purs` (see https://github.com/input-output-hk/cardano-sl/blob/develop/explorer/frontend/src/Explorer/Update.purs#L487), which occurs that error. Wrapping it around to not using a foreign directly fixed this error.